### PR TITLE
Implement /joined_rooms

### LINKED ISSUE #605: Implement /joined_rooms
- [ ] Implement [GET /_matrix/client/r0/joined_rooms](https://matrix.org/docs/spec/client_server/r0.4.0.html#get-matrix-client-r0-joined-rooms)

### PR DESCRIPTION
**Pull request by prateek2211**
fixed #605 

### Pull Request Checklist

* [X] Implement GET /_matrix/client/r0/joined_rooms
* [X] I have added any new tests that need to pass to `testfile` as specified in [docs/sytest.md](https://github.com/matrix-org/dendrite/blob/master/docs/sytest.md)
* [X] Pull request includes a [sign off](https://github.com/matrix-org/dendrite/blob/master/CONTRIBUTING.md#sign-off)

Signed-off-by: Prateek Sachan psachan@cs.iitr.ac.in

### CODE REVIEW

File: clientapi/routing/routing.go [hunk 1]
```diff
@@ -355,6 +361,12 @@ func Setup(
 		}),
 	).Methods(http.MethodGet, http.MethodOptions)
 
+	r0mux.Handle("/account/deactivate",
+		common.MakeAuthAPI("account_deactivate", authData, func(req *http.Request, device *authtypes.Device) util.JSONResponse {
+			return GetAssociated3PIDs(req, accountDB, device)
+		}),
+	).Methods(http.MethodPost, http.MethodOptions)
+
```
Comment (Kegsay) [inline clientapi/routing/routing.go]: Why is this in this PR?

**Comment by Kegsay** [comment]
Otherwise LGTM

File: clientapi/routing/routing.go (hunk 1 above)
Comment (prateek2211) [reply clientapi/routing/routing.go]: oops! corrected it now

**Comment by Kegsay** [comment]
Sorry to be a pain, but can you please add the `/joined_rooms` tests to `sytest-whitelist` so we can let sytest test this endpoint? Currently:
 - `Test 65 GET /joined_rooms lists newly-created room... EXPECTED FAIL`
 - `Test 295 /joined_rooms returns only joined rooms... SKIP`

You need to add those exact test names "GET /joined_rooms lists newly-created room" and "/joined_rooms returns only joined rooms" to the whitelist.

295 is skipped because 65 fails. If you click on the Artifacts for these tests and download `results.tap` you can see:
```
not ok 65 (expected fail) GET /joined_rooms lists newly-created room # TODO expected fail
# Started: 2020-03-18 12:02:15.401
# Ended: 2020-03-18 12:02:15.404
# HTTP Request failed ( 404 Not Found https://localhost:8800/_matrix/client/unstable/joined_rooms?access_token=3-RhAigrUJoFdrV_O2WlwKekeAE8uwVee8unvR8iuo4 )
```
so for some reason this endpoint isn't hooked up? Once this is done and passing I'll merge it in. Great job!

**Comment by prateek2211** [comment]
yeah sure

**Comment by Kegsay** [comment]
Right, I see that synapse supports r0/v1/unstable for this endpoint, and sytest uses `unstable`. It's in `r0` of the spec though, so I'll modify sytest to use the `r0` endpoint. Bear with me...

**Comment by Kegsay** [comment]
FYI: https://github.com/matrix-org/sytest/pull/834

**Comment by prateek2211** [comment]
Yay, checks passed!!

**Pull request merged**